sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/object.scala

```diff
@@ -41,7 +41,20 @@ object CatalystSerde {
   }
 
   def generateObjAttr[T : Encoder]: Attribute = {
-    AttributeReference("obj", encoderFor[T].deserializer.dataType, nullable = false)()
+    val deserializer = encoderFor[T].deserializer
+    val dataType = deserializer.dataType
+    val nullable = if (deserializer.childrenResolved) {
```

Comment:
I made a mistake. I should use `val nullable = if (deserializer.resolved) {`. Even when I made this correction, `deserializer` is resolved.

In the above context, it is ensured that `encoderFor[T].deserializer` is resolved in the assertion `require(inputAttributes.forall(_.resolved), "Input attributes must all be resolved.")` at  `case class UnresolvedDeserializer`.